v-next/hardhat-utils/src/hex.ts

```diff
@@ -95,7 +95,17 @@ export function hexStringToNumber(hexString: string): number {
  * @returns PrefixedHexString The hexadecimal representation of the bytes.
  */
 export function bytesToHexString(bytes: Uint8Array): PrefixedHexString {
-  return `0x${Buffer.from(bytes).toString("hex")}`;
+  return bufferToHexString(Buffer.from(bytes));
+}
+
+/**
+ * Converts a Buffer to a hexadecimal string.
+ *
+ * @param buffer The buffer to convert.
+ * @returns PrefixedHexString The hexadecimal representation of the buffer.
+ */
+export function bufferToHexString(buffer: Buffer): PrefixedHexString {
```

Comment:
I think we don't need this. It may not be entirely obvious, but `Buffer` extends `Uint8Array`, so you can just use `bytesToHexString`.

`hardhat-utils` intentionally tries not to have Node types in its interface.

Reply:
Ah, OK, I understand now. Thanks, I'll revert that part of the PR and use bytesToHexString instead.